src/marlin_pc/data_structures.rs

```diff
@@ -143,6 +146,64 @@ impl<E: PairingEngine> ToBytes for VerifierKey<E> {
     }
 }
 
+/// `PreparedVerifierKey` is used to check evaluation proofs for a given commitment.
+#[derive(Derivative)]
+#[derivative(Default(bound = ""), Clone(bound = ""), Debug(bound = ""))]
+pub struct PreparedVerifierKey<E: PairingEngine> {
+    /// The verification key for the underlying KZG10 scheme.
+    pub prepared_vk: kzg10::PreparedVerifierKey<E>,
+    /// Information required to enforce degree bounds. Each pair
+    /// is of the form `(degree_bound, shifting_advice)`.
+    /// This is `None` if `self` does not support enforcing any degree bounds.
+    pub prepared_degree_bounds_and_shift_powers: Option<Vec<(usize, Vec<E::G1Affine>)>>,
+    /// The maximum degree supported by the `UniversalParams` `self` was derived
+    /// from.
+    pub max_degree: usize,
+    /// The maximum degree supported by the trimmed parameters that `self` is
+    /// a part of.
+    pub supported_degree: usize,
```

Comment:
Invoking `Default::default()` here will set the fields to be empty/zero. Are you sure that's the intended behaviour?

Reply:
@weikengchen Do you know about this as well?

Reply:
Let me double-check---especially if we use the prepared verifier key's default somewhere.

Reply:
Remove the Default trait and the derivation.